fastlane/Fastfile

```diff
@@ -424,16 +355,125 @@ platform :android do
     version = release_version_current
 
     # Wrap up
-    remove_branch_protection(repository: GH_REPOSITORY, branch: "release/#{version}")
-    set_milestone_frozen_marker(repository: GH_REPOSITORY, milestone: version, freeze: false)
-    create_new_milestone(repository: GH_REPOSITORY, need_appstore_submission: true)
-    close_milestone(repository: GH_REPOSITORY, milestone: version)
+    remove_branch_protection(repository: GITHUB_REPO, branch: "release/#{version}")
+    set_milestone_frozen_marker(repository: GITHUB_REPO, milestone: version, freeze: false)
+    create_new_milestone(repository: GITHUB_REPO, need_appstore_submission: true)
+    close_milestone(repository: GITHUB_REPO, milestone: version)
 
     push_to_git_remote(tags: false)
     trigger_release_build(branch_to_build: "release/#{version}")
     create_backmerge_pr
   end
 
+  # This lane publishes a release on GitHub and creates a PR to backmerge the current release branch into the next release/ branch
+  #
+  # @param [Boolean] skip_confirm (default: false) If set, will skip the confirmation prompt before running the rest of the lane
+  #
+  lane :publish_release do |skip_confirm: false|
+    ensure_git_status_clean
+    ensure_git_branch(branch: '^release/')
+
+    version_number = release_version_current
+
+    current_branch = "release/#{version_number}"
+    next_release_branch = "release/#{release_version_next}"
+
+    UI.important <<~PROMPT
+      Publish the #{version_number} release. This will:
+      - Publish the existing draft `#{version_number}` release on GitHub
+      - Which will also have GitHub create the associated git tag, pointing to the tip of the branch
+      - If the release branch for the next version `#{next_release_branch}` already exists, backmerge `#{current_branch}` into it
+      - If needed, backmerge `#{current_branch}` back into `#{DEFAULT_BRANCH}`
+      - Delete the `#{current_branch}` branch
+    PROMPT
+    UI.user_error!("Terminating as requested. Don't forget to run the remainder of this automation manually.") unless skip_confirm || UI.confirm('Do you want to continue?')
+
+    UI.important "Publishing release #{version_number} on GitHub"
+
+    publish_github_release(
+      repository: GITHUB_REPO,
+      name: version_number
+    )
+
+    create_backmerge_pr
```

Comment:
FYI, we have seen issues when creating a Backmerge PR at this stage in the release process in a repo with squash merge enabled (which I realized is the case for PCAndroid), as this PR will show all the release branch changes that have already been merged. See related PR on WPAndroid: https://github.com/wordpress-mobile/WordPress-Android/pull/21535

Reply:
Ah damn 😔 Good point.

I wonder if there's a way in git to detect that… not sure given squash commits by definition lose history and create entirely new sha… (one more reason why I'm not a fan of those, but that's another debate 😅)

The Release Scenario currently have a manual task about creating a backmerge PR at that stage already iirc though… so we'll have to look at what the manual backmerge PRs they created manually so far in previous releases look like?

And in any case we will at the very least need to still do a backmerge PR to the next `release/*` branch if not to the default one.

Reply:
> And in any case we will at the very least need to still do a backmerge PR to the next release/* branch if not to the default one.

True -- though doing it using the shared action `create_release_backmerge_pull_request_action` will have the same issue, I believe.